powerauth-java-server/src/test/java/com/wultra/powerauth/fido2/Fido2AuthenticatorTest.java

```diff
@@ -0,0 +1,440 @@
+import com.webauthn4j.data.extension.client.RegistrationExtensionClientOutput;
+import com.webauthn4j.test.EmulatorUtil;
+import com.webauthn4j.test.authenticator.webauthn.SelfAttestedPackedAuthenticator;
+import com.webauthn4j.test.authenticator.webauthn.WebAuthnAuthenticatorAdaptor;
+import com.webauthn4j.test.client.ClientPlatform;
+import com.wultra.powerauth.fido2.errorhandling.Fido2AuthenticationFailedException;
+import com.wultra.powerauth.fido2.rest.model.entity.*;
+import com.wultra.powerauth.fido2.rest.model.request.AssertionChallengeRequest;
+import com.wultra.powerauth.fido2.rest.model.request.AssertionVerificationRequest;
+import com.wultra.powerauth.fido2.rest.model.request.RegistrationRequest;
+import com.wultra.powerauth.fido2.rest.model.response.AssertionChallengeResponse;
+import com.wultra.powerauth.fido2.rest.model.response.AssertionVerificationResponse;
+import com.wultra.powerauth.fido2.rest.model.response.RegistrationChallengeResponse;
+import com.wultra.powerauth.fido2.rest.model.response.RegistrationResponse;
+import com.wultra.powerauth.fido2.service.AssertionService;
+import com.wultra.powerauth.fido2.service.RegistrationService;
+import com.wultra.security.powerauth.client.model.enumeration.ActivationStatus;
+import com.wultra.security.powerauth.client.model.enumeration.SignatureType;
+import com.wultra.security.powerauth.client.model.request.CreateApplicationRequest;
+import com.wultra.security.powerauth.client.model.request.GetActivationStatusRequest;
+import com.wultra.security.powerauth.client.model.request.OperationTemplateCreateRequest;
+import com.wultra.security.powerauth.client.model.response.GetApplicationListResponse;
+import com.wultra.security.powerauth.client.model.response.OperationTemplateDetailResponse;
+import com.wultra.security.powerauth.client.model.response.OperationTemplateListResponse;
+import io.getlime.security.powerauth.app.server.Application;
+import io.getlime.security.powerauth.app.server.service.PowerAuthService;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test of self-attested packed authenticator against PowerAuth server.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@SpringBootTest(classes = Application.class)
+@ExtendWith(SpringExtension.class)
+public class Fido2AuthenticatorTest {
+
+    private final CBORMapper CBOR_MAPPER = new CBORMapper();
+    private final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private final static String RP_ID = "powerauth.com";
```

Comment:
May we use `RELYING_PARTY_ID` instead of `RP_ID`, please?

Reply:
No, this is how it is in the specification.